src/backend/cgen.c

```diff
@@ -27,6 +27,14 @@
 static char __file__[] = __FILE__;      /* for tassert.h                */
 #include        "tassert.h"
 
+/*********************************
+ */
+CodeBuilder::CodeBuilder(code *c)
+{
+    head = c;
+    pTail = c ? &code_last(c)->next : &head;
```

Comment:
Basically, that constructor in DtBuilder can be replaced with a null initialiazer, at the cost of having a test in the other member functions. This is not a big deal or any sort of blocker for conversion to D. The beauty of it is, as I said, the fact that it is private and encapsulated such that **none** of the DtBuilder users need to be aware of such a change.